### ISSUE TITLE
Num.arange only accepts Int32's

### ISSUE THREAD

**Issue by Iainmon** [crystal-data/num.cr#19]
https://github.com/crystal-data/num.cr/blob/32a5d0701dd7cef3485867d2afd897900ca60901/src/tensor/creation.cr#L159

Shouldn't it be 
```crystal
def arange(start : Number, stop : Number, step : Number = 1)
```
?

I'm kinda new to crystal, so I wasn't super sure what `forall U` was.

**Comment by Iainmon** [comment]
This fails:
```crystal
def arange(start, stop, step)
    diff = stop - start
    count = diff.abs / step
    Num.linspace(start, stop, count)
end
```

**Comment by christopherzimmerman** [comment]
Yes, it should definitely be something like that.  I don't think it's even necessary to enforce integer, even though linspace is usually better for non-integer steps.  The `forall U` specifies a generic function definition, check out the section called [Free Variables](https://crystal-lang.org/reference/syntax_and_semantics/type_restrictions.html).

I would welcome a PR here!

**Comment by christopherzimmerman** [comment]
Closed by #21
